c++/src/kj/async-inl.h

```diff
@@ -1093,13 +1093,17 @@ inline PromiseForResult<Func, void> evalNow(Func&& func) {
 
 template <typename Func>
 inline PromiseForResult<Func, WaitScope&> startFiber(size_t stackSize, Func&& func) {
-  typedef _::FixVoid<_::ReturnType<Func, WaitScope&>> ResultT;
-
-  Own<_::FiberBase> intermediate = kj::heap<_::Fiber<Func>>(stackSize, kj::fwd<Func>(func));
-  intermediate->start();
-  auto result = _::PromiseNode::to<_::ChainPromises<_::ReturnType<Func, WaitScope&>>>(
-      _::maybeChain(kj::mv(intermediate), implicitCast<ResultT*>(nullptr)));
-  return _::maybeReduce(kj::mv(result), false);
+  #if KJ_NO_EXCEPTIONS
+    KJ_UNIMPLEMENTED("Fibers require exceptions");
```

Comment:
I'd still like to see this moved into FiberBase's constructor. A couple issues with having it here:

* It's a little messy to have two different places in the code where we're using #ifdef's to check whether fibers are supported and throwing if not. Might as well consolidate that.
* Generally `kj/debug.h` isn't supposed to be included from other headers, so this macro shouldn't be available here... apparently it must have been included by mistake. The motivation behind that rule is that I worry that these macros expand into a surprising amount of code and therefore putting them into inlined methods and templates could contribute to excessive code bloat.